packages/headless/src/features/search-parameters/search-parameter-serializer.ts

```diff
@@ -172,9 +178,22 @@ function processObjectValues(key: string, values: string[]) {
 
 function buildNumericRanges(ranges: string[]) {
   return ranges
-    .map((str) => str.split(rangeDelimiter).map(parseFloat))
-    .filter((range) => range.length === 2 && range.every(Number.isFinite))
-    .map(([start, end]) => buildNumericRange({start, end, state: 'selected'}));
+    .map((str) => {
+      const isEndInclusive = str.indexOf(rangeDelimiterInclusive) !== -1;
+      const [startAsString, endAsString] = str.split(
+        isEndInclusive ? rangeDelimiterInclusive : rangeDelimiterExclusive
+      );
```

Comment:
This bit of code is repeated between date and numeric, do you think we should refactor it?